### PR TITLE
Refactor calibration, clock, and sercom modules for SAMD51/SAME54

### PR DESCRIPTION
**Pull request by jessebraham**
These two device families are essentially identical other than a handful of small difference, all of which can be handled with `#[cfg()]`. This pulls out the `calibration`, `clock`, and `sercom` modules into the `common::thumbv7em` module.

### CODE REVIEW

**Comment by jessebraham** [comment]
@mvniekerk would you be able to test this out on the SAME54 by any chance?

File: hal/src/common/thumbv7em/sercom/uart.rs [hunk 1]
```diff
@@ -156,7 +156,8 @@ macro_rules! uart {
                             w.runstdby().set_bit(); // Run in standby
                             w.form().bits(0); // 0 is no parity bits
 
-                            // w.mode().usart_int_clk(); // Internal clock mode
+                            // TODO: does this need to be commented out to work on same54?
+                            w.mode().usart_int_clk(); // Internal clock mode
```
Comment (jessebraham) [inline hal/src/common/thumbv7em/sercom/uart.rs:160]: This line was commented out in the SAME54 implementation, but not in the SAMD51 implementation. Any ideas on why this was this case?

**Comment by Sympatron** [comment]
Is something blocking this PR (apart from the merge conflict, that is not that hard to resolve)? I think it is really helpful and reduces code duplication a lot without breaking anything.
I wanted to write a PR that adds more variants of the SAME series, but that would be a PITA without this PR.

**Comment by jessebraham** [comment]
I was hoping to get a test on the SAME54 for the one line mentioned above, and then forgot about it 😁 @sajattack once I rebase this are you okay with merging it as-is?

**Comment by jessebraham** [comment]
Okay great, thanks. I'll try to get this done this weekend. There has been a new PAC added since I opened this PR, so unfortunately it's going to be a bit more involved than just rebasing, but still shouldn't take too long.

**Comment by jessebraham** [comment]
I managed to hammer it out during my lunch break, I think it should be fine but I have yet to test on actual hardware. Will do that after work.

**Comment by twitchyliquid64** [comment]
Looks good at first pass. Also spotchecked SERCOM5 and LGTM.

**Pull request merged**